klog.go

```diff
@@ -793,23 +775,52 @@ func (l *loggingT) output(s severity, buf *buffer, file string, line int, alsoTo
 		// Write the stack trace for all goroutines to the files.
 		trace := stacks(true)
 		logExitFunc = func(error) {} // If we get a write error, we'll still exit below.
-		for log := fatalLog; log >= infoLog; log-- {
-			if f := l.file[log]; f != nil { // Can be nil if -logtostderr is set.
-				f.Write(trace)
-			}
-		}
+		l.writeToDifferentSeverity(fatalLog, trace)
 		l.mu.Unlock()
 		timeoutFlush(10 * time.Second)
 		os.Exit(255) // C++ uses -1, which is silly because it's anded with 255 anyway.
 	}
 	l.putBuffer(buf)
 	l.mu.Unlock()
+	// Note: The stats estimate logs for each severity level individually,
+	// even in the situation that log-file is specified and
+	// all severity-level logs are tracked only in the infoLog file.
 	if stats := severityStats[s]; stats != nil {
 		atomic.AddInt64(&stats.lines, 1)
 		atomic.AddInt64(&stats.bytes, int64(len(data)))
 	}
 }
 
+//writeToDifferentSeverity writes |data| to the `s` and its lower severity file.
+// e.g. If Severity level is Error, the data will be written to all the Error,
+// Warning, and Info log file. However, if --log_file flag is provided, klog
+// no longer tracks logs separately due to their severity level, but rather
+// only write to the info logfile which later on will be flushed to the
+// dedicated log_file.
+func (l *loggingT) writeToDifferentSeverity(s severity, data []byte) {
```

Comment:
I recommend factoring this a bit differently:

Make the function:
```go
func (l *logging.T) writeLogData(s severity, data byte) {
  if l.shouldLogToStderr(s) { ... }
  if l.logFile != "" {
    ... // handle single-file mode
  } else {
    ... // handle multi-file mode
  }
}
```

Reply:
Sure. but it will have many duplicate code. Is that okay?

Reply:
I don't think it will have too much. I'm worried that the current approach of defaulting to the info log is too error-prone, and folding in the log-to-stderr functionality will actually deduplicate code (and fix a bug) in the fatal log path.